### ISSUE TITLE
[SPIKE] Use python 3 across all distributions

### ISSUE THREAD

**Issue by atsikham** [epiphany-platform/epiphany#1593]
**Is your feature request related to a problem? Please describe.**
Currently we use an outdated python2 version that is not supported. Moreover, python versions across supported OS distributions are different. As a result some ansible modules cannot be used as it's not possible to install necessary dist packages. Probably it will cause other issues in future.
Example: `python-cryptography` required for `openssl_*` ansible modules cannot be installed on ubuntu 18.04:
```text
The following packages have unmet dependencies:
 python-cryptography : Depends: python-asn1crypto but it is not installable
                       Depends: python-cffi-backend-api-min (<= 9729) but it is not installable
                       Depends: python-cffi-backend-api-max (>= 9729) but it is not installable
                       Depends: python-enum34 but it is not installable
                       Depends: python-idna (>= 2.1) but it is not installable
                       Depends: python-ipaddress but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Describe the solution you'd like**
Python3 is used on all target hosts for Ansible.

**Describe alternatives you've considered**
No

**Additional context**
Need to verify python dependencies and test that after switching all/main tasks work correctly.

---

**DoD checklist**

* [x] Reader is able to understand the results of spike
* [ ] The results of the spike are presented in a table (to show simply what are compared or researched parameters) / not applicable
* [ ] Each value / cell in the results table is described more deeply below
* [ ] Demo of the spike (automated as much as possible)
* [ ] Design doc updated

**Comment by atsikham** [comment]
The question is what kind of tests should be performed after python version upgrade.

**Comment by seriva** [comment]
I think we can review and close this after we are done with RedHat upgrade.

**Comment by to-bar** [comment]
Python 3.6 is present on RHEL 8 out of the box (as `/usr/libexec/platform-python`). We also install `python3` package (Python 3.6.8).
This task can be closed after #1332 is done.
